crates/red_knot_python_semantic/resources/mdtest/generics/functions.md

```diff
@@ -162,61 +147,39 @@ parameters simultaneously.
 def two_params[T](x: T, y: T) -> T:
     return x
 
-# TODO: no error
 # TODO: revealed: str
-# error: [invalid-argument-type]
-# error: [invalid-argument-type]
-reveal_type(two_params("a", "b"))  # revealed: T
+reveal_type(two_params("a", "b"))  # revealed: Literal["a", "b"]
 
-# TODO: no error
 # TODO: revealed: str | int
-# error: [invalid-argument-type]
-# error: [invalid-argument-type]
-reveal_type(two_params("a", 1))  # revealed: T
+reveal_type(two_params("a", 1))  # revealed: Literal["a", 1]
 ```
 
 ```py
 def param_with_union[T](x: T | int, y: T) -> T:
     return y
 
-# TODO: no error
 # TODO: revealed: str
-# error: [invalid-argument-type]
-reveal_type(param_with_union(1, "a"))  # revealed: T
+reveal_type(param_with_union(1, "a"))  # revealed: Literal["a"]
 
-# TODO: no error
 # TODO: revealed: str
-# error: [invalid-argument-type]
-# error: [invalid-argument-type]
-reveal_type(param_with_union("a", "a"))  # revealed: T
+reveal_type(param_with_union("a", "a"))  # revealed: Literal["a"]
 
-# TODO: no error
 # TODO: revealed: int
-# error: [invalid-argument-type]
-reveal_type(param_with_union(1, 1))  # revealed: T
+reveal_type(param_with_union(1, 1))  # revealed: Literal[1]
```

Comment:
Interesting case not covered here: `param_with_union(3, 1)` should be a valid call, and must return `Literal[1]`, not `Literal[3]`, for the same reason discussed above.

Reply:
Yep! Added